### PR TITLE
feat(drive)!: fees distribution

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Integrate fee distribution logic implemented in https://github.com/dashevo/rs-drive/pull/105

## What was done?

- Disabled fee validation logic since it's not ready in RS Drive
- Call RS Drive ABCI handlers
- Fixed synchronize masternode identities logging
- Removed accumulative fee pool

## How Has This Been Tested?

With tests

## Breaking Changes

Blockchain data and state are changed

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/js-drive/lib/abci/handlers/beginBlockHandlerFactory.js [hunk 1]
```diff
@@ -119,6 +122,30 @@ function beginBlockHandlerFactory(
...
 
+    // Call RS ABCI
+
+    /**
+     * @type {BlockBeginRequest}
+     */
+    const rsRequest = {
+      blockHeight: header.height.toNumber(),
+      blockTime: timeToMillis(header.time.seconds, header.time.nanos),
+      proposerProTxHash: Array.from(header.proposerProTxHash),
```
Comment (shumkov) [inline packages/js-drive/lib/abci/handlers/beginBlockHandlerFactory.js:133]: Now we can pass Buffer

File: packages/js-drive/lib/abci/handlers/beginBlockHandlerFactory.js [hunk 2]
```diff
@@ -138,11 +165,12 @@ function beginBlockHandlerFactory(
         `Masternode identities are synced for heights from ${fromHeight} to ${toHeight}: ${createdEntities.length} created, ${updatedEntities.length} updated, ${removedEntities.length} removed`,
       );
 
+      // TODO: figure out why some of the identities is null
       consensusLogger.trace(
         {
-          createdEntities: createdEntities.map((item) => item.toJSON()),
-          updatedEntities: updatedEntities.map((item) => item.toJSON()),
-          removedEntities: removedEntities.map((item) => item.toJSON()),
+          createdEntities: createdEntities.filter((i) => i !== null).map((item) => item.toJSON()),
```
Comment (shumkov) [inline packages/js-drive/lib/abci/handlers/beginBlockHandlerFactory.js:171]: We need to figure out why we get null. It shouldn't be the case.

File: packages/js-drive/lib/abci/handlers/deliverTxHandlerFactory.js [hunk 3]
```diff
@@ -144,27 +144,28 @@ function deliverTxHandlerFactory(
...
-    }
+    // if (updatedBalance < 0) {
+    //   throw new NegativeBalanceError(identity);
+    // }
 
-    await transactionalDpp.getStateRepository().updateIdentity(identity);
+    // await transactionalDpp.getStateRepository().updateIdentity(identity);
 
-    blockExecutionContext.incrementCumulativeFees(actualStateTransitionFee);
+    blockExecutionContext.incrementCumulativeProcessingFee(actualStateTransitionFee);
```
Comment (shumkov) [inline packages/js-drive/lib/abci/handlers/deliverTxHandlerFactory.js:168]: If you look bellow you will find:

```js
    const {
      storageFee: actualStorageFee,
      processingFee: actualProcessingFee,
    } = calculateOperationFees(actualStateTransitionOperations);
```

That's what you need to increment both processing and storage fees

File: packages/js-drive/lib/abci/handlers/endBlockHandlerFactory.js [hunk 4]
```diff
@@ -54,6 +61,56 @@ function endBlockHandlerFactory(
...
+      processingFees,
+      storageFees,
+    }, `${processingFees} processing fees add to epoch #${currentEpochIndex}. ${storageFees} storage fees add to distribution pool`);
+
+    if (rsResponse.masternodesPaidCount) {
+      consensusLogger.debug({
+        currentEpochIndex,
+        masternodesPaidCount: rsResponse.masternodesPaidCount,
+        paidEpochIndex: rsResponse.paidEpochIndex,
+      }, `${rsResponse.masternodesPaidCount} masternodes paid for epoch #${rsResponse.paidEpochIndex}`);
```
Comment (shumkov) [inline packages/js-drive/lib/abci/handlers/endBlockHandlerFactory.js:111]: "masternodes were paid"?

File: packages/platform-test-suite/test/functional/platform/Identity.spec.js [hunk 5]
```diff
@@ -299,7 +299,8 @@ describe('Platform', () => {
       });
     });
 
-    describe('Credits', () => {
+    // TODO: enable once fee calculation is done
+    xdescribe('Credits', () => {
```
Comment (shumkov) [inline packages/platform-test-suite/test/functional/platform/Identity.spec.js:303]: We usually use `describe.skip`

File: packages/js-drive/lib/identity/masternode/handleNewMasternodeFactory.js [hunk 6]
```diff
@@ -79,13 +79,11 @@ function handleNewMasternodeFactory(
...
 
       // Create a document in rewards data contract with percentage
-      result.push(
-        await createRewardShareDocument(
-          dataContract,
-          masternodeIdentifier,
-          operatorIdentifier,
-          proRegTxPayload.operatorReward,
-        ),
+      await createRewardShareDocument(
```
Comment (shumkov) [inline packages/js-drive/lib/identity/masternode/handleNewMasternodeFactory.js:82]: Shouldn't we add this to the result if the returned value is not null?

Comment (jawid-h) [reply packages/js-drive/lib/identity/masternode/handleNewMasternodeFactory.js:82]: Why? It's not an identity.

File: packages/platform-test-suite/test/functional/platform/Identity.spec.js (hunk 5 above)
Comment (shuplenkov) [inline packages/platform-test-suite/test/functional/platform/Identity.spec.js:303]: `xdescribe` ?

**Review by jawid-h — DISMISSED** [review]

File: packages/platform-test-suite/test/functional/platform/Identity.spec.js (hunk 5 above)
Comment (jawid-h) [reply packages/platform-test-suite/test/functional/platform/Identity.spec.js:303]: Yeah, to disable tests

**Review by shuplenkov — APPROVED** [review]

**Pull request merged**